File: src/main/java/io/junseok/mealmate/domain/board/repository/BoardRepository.java
package io.junseok.mealmate.domain.board.repository;

import io.junseok.mealmate.domain.board.entity.Board;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface BoardRepository extends JpaRepository<Board,Long> {
    List<Board> findAllByOrderByCreateDtDesc();
}

File: src/main/java/io/junseok/mealmate/domain/board/repository/BoardRepository.kt
package io.junseok.mealmate.domain.board.repository;

import io.junseok.mealmate.domain.board.entity.Board;
import io.junseok.mealmate.domain.restaurant.entity.Restaurant
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository;

interface BoardRepository : JpaRepository<Board,Long> {
    fun findAllByOrderByCreateDtDesc(pageable: Pageable): Page<Board>

    fun findAllByRestaurantOrderByCreateDtDesc(pageable: Pageable,restaurant: Restaurant):Page<Board>
}
